src/main/kotlin/com/pubnub/api/models/consumer/access_manager/sum/UserIdGrant.kt

```diff
@@ -0,0 +1,45 @@
+package com.pubnub.api.models.consumer.access_manager.sum
+
+import com.pubnub.api.PubNubError
+import com.pubnub.api.PubNubException
+import com.pubnub.api.UserId
+import com.pubnub.api.models.consumer.access_manager.v3.PNGrant
+import com.pubnub.api.models.consumer.access_manager.v3.PNUUIDPatternGrant
+import com.pubnub.api.models.consumer.access_manager.v3.PNUUIDResourceGrant
+import com.pubnub.api.models.consumer.access_manager.v3.UUIDGrant
+
+interface UserIdGrant : PNGrant {
+    companion object {
+        fun id(
+            userId: UserId,
+            get: Boolean = false,
+            update: Boolean = false,
+            delete: Boolean = false
+        ): UserIdGrant = PNUUIDResourceGrant(
+            id = userId.value,
+            delete = delete,
+            get = get,
+            update = update
+        )
+
+        fun pattern(
+            pattern: String,
+            get: Boolean = false,
+            update: Boolean = false,
+            delete: Boolean = false
+        ): UserIdGrant = PNUUIDPatternGrant(
+            id = pattern,
+            delete = delete,
+            get = get,
+            update = update
+        )
+    }
+}
+
+fun UserIdGrant.toUuidGrants(): UUIDGrant {
+    return when (this) {
+        is PNUUIDResourceGrant -> PNUUIDResourceGrant(userIdGrant = this)
```

Comment:
What? I have trouble understanding this line. If we have `PNUUIDResourceGrant` then we'll create `PNUUIDResourceGrant` passing this thing as an argument. But why? We already have `PNUUIDResourceGrant`, so what's the point?